db-repo/include/emeralddb/core.hpp

```diff
@@ -0,0 +1,72 @@
+/*    Copyright 2009 10gen Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+*/
+
+#ifndef CORE_HPP__
+#define CORE_HPP__
+
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <sys/time.h>
+#include <time.h>
+#include <stdarg.h>
+#include <unistd.h>
+//#include <syscall.h>
```

Comment:
local dev environment mac does not `syscall.h` this header file, here needs add some macros to check deploy environment and support different lib's import like win, linux and mac